tests/Test/Xrefcheck/IgnoringSpec.hs

```diff
@@ -0,0 +1,50 @@
+{- SPDX-FileCopyrightText: 2019 Serokell <https://serokell.io>
+ -
+ - SPDX-License-Identifier: MPL-2.0
+ -}
+
+module Test.Xrefcheck.IgnoringSpec where
+
+import qualified Data.ByteString.Lazy as BSL
+import Test.Hspec (Spec, describe, it, shouldBe)
+
+import Xrefcheck.Core
+import Xrefcheck.Scanners.Markdown
+
+spec :: Spec
+spec = do
+    describe "Parsing failures" $ do
+        it "Check if parsing incorrect markdowns produce exceptions" $ do
+            areIncorrect <- mapM isIncorrectMD failPaths
+            any id areIncorrect `shouldBe` True
+    describe "\"ignore link\" mode" $ do
+        it "Check \"ignore link\" performance" $ do
+            fi <- getFI "tests/markdowns/ignore_link.md"
+            (getRefs fi) `shouldBe` ["team"]
+    describe "\"ignore paragraph\" mode" $ do
+        it "Check \"ignore paragraph\" performance" $ do
+            fi <- getFI "tests/markdowns/ignore_paragraph.md"
+            (getRefs fi) `shouldBe` ["blog", "contacts"]
+    describe "\"ignore file\" mode" $ do
+        it "Check \"ignore file\" performance" $ do
+            fi <- getFI "tests/markdowns/ignore_file.md"
+            (getRefs fi) `shouldBe` []
+    where
+        failPaths =
+            [ "tests/markdowns/no_link.md"
+            , "tests/markdowns/no_paragraph.md"
+            , "tests/markdowns/unexpected_ignore_file.md"
+            , "tests/markdowns/unrecognised_option.md"
+            ]
+        
+        defFI = FileInfo [] []
+
+        parse path = parseFileInfo . decodeUtf8 <$> BSL.readFile path
+        
+        getFI path = fromRight defFI <$> parse path
```

Comment:
It's better to just fail in such cases in tests, otherwise, if this error hits, investigating what exactly is going on may take a while :)  And a failure is quite unexpected here, so we can afford failing.